src/app/core/services/asset-entry-builder.ts

```diff
@@ -1,27 +1,31 @@
 import {Injectable} from '@angular/core';
-import {AssetEditorDialogFormValue} from '../../routes/connector-ui/asset-page/asset-create-dialog/asset-editor-dialog-form-model';
-import {AssetEntryDto} from './api/legacy-managent-api-client';
+import {AssetCreateRequest} from '@sovity.de/edc-client/dist/generated/models/AssetCreateRequest';
+import {
+  AssetEditorDialogFormValue
+} from '../../routes/connector-ui/asset-page/asset-create-dialog/asset-editor-dialog-form-model';
 import {AssetPropertyMapper} from './asset-property-mapper';
 import {DataAddressMapper} from './data-address-mapper';
 
+
 @Injectable()
 export class AssetEntryBuilder {
   constructor(
     private assetPropertyMapper: AssetPropertyMapper,
     private dataAddressMapper: DataAddressMapper,
-  ) {}
+  ) {
+  }
 
   /**
-   * Build {@link AssetEntryDto} from {@link AssetEditorDialogFormValue}
+   * Build {@link AssetCreateRequest} from {@link AssetEditorDialogFormValue}
    *
    * @param formValue form value
    * @return asset create dto
    */
-  buildAssetEntry(formValue: AssetEditorDialogFormValue): AssetEntryDto {
+  buildAssetEntry(formValue: AssetEditorDialogFormValue): AssetCreateRequest {
     let properties = this.assetPropertyMapper.buildProperties(formValue);
-    const dataAddress = this.dataAddressMapper.buildDataAddressProperties(
+    const dataAddressProperties = this.dataAddressMapper.buildDataAddressProperties(
```

Comment:
careful, I think buildDataAddressProperties builds a DataAddressDto which is `{properties: {'a': 'b', ...}}`. The uses of that DataAddressDto should be removed for this code branch.

If the method is called buildDataAddressProperties, you should rebuild it so it will return a `Record<String, String>`, not the `DataAddressDto`